contracts/contracts/validator-registry/avs/MevCommitAVS.sol

```diff
@@ -580,12 +587,15 @@ contract MevCommitAVS is IMevCommitAVS, MevCommitAVSStorage,
     function _isValidatorOptedIn(bytes calldata valPubKey) internal view returns (bool) {
         bool isValRegistered = validatorRegistrations[valPubKey].exists;
         bool isFrozen = validatorRegistrations[valPubKey].freezeHeight.exists;
-        bool isDeregRequested = validatorRegistrations[valPubKey].deregRequestHeight.exists;
+        bool isValDeregRequested = validatorRegistrations[valPubKey].deregRequestHeight.exists;
         IEigenPod pod = _eigenPodManager.getPod(validatorRegistrations[valPubKey].podOwner);
         bool isValActive = pod.validatorPubkeyToInfo(valPubKey).status == IEigenPod.VALIDATOR_STATUS.ACTIVE;
         address delegatedOperator = _delegationManager.delegatedTo(validatorRegistrations[valPubKey].podOwner);
         bool isOperatorRegistered = operatorRegistrations[delegatedOperator].exists;
-        return isValRegistered && !isFrozen && !isDeregRequested && isValActive && isOperatorRegistered;
+        bool isOperatorDeregRequested = operatorRegistrations[delegatedOperator].deregRequestHeight.exists;
+
+        return isValRegistered && !isFrozen && !isValDeregRequested && isValActive
+            && isOperatorRegistered && !isOperatorDeregRequested;
```

Comment:
To minimize the number of storage reads I would suggest to refactor this function:

```solidity
    ValidatorRegistration memory valRegistration = validatorRegistrations[valPubKey];
    bool isValRegistered = valRegistration.exists;
    bool isFrozen = valRegistration.freezeHeight.exists;
    bool isValDeregRequested = valRegistration.deregRequestHeight.exists;
    
    IEigenPod pod = _eigenPodManager.getPod(valRegistration.podOwner);
    bool isValActive = pod.validatorPubkeyToInfo(valPubKey).status == IEigenPod.VALIDATOR_STATUS.ACTIVE;
    
    address delegatedOperator = _delegationManager.delegatedTo(valRegistration.podOwner);
    OperatorRegistration memory operatorRegistration = operatorRegistrations[delegatedOperator];
    bool isOperatorRegistered = operatorRegistration.exists;
    bool isOperatorDeregRequested = operatorRegistration.deregRequestHeight.exists;

    return isValRegistered && !isFrozen && !isValDeregRequested && isValActive
        && isOperatorRegistered && !isOperatorDeregRequested;

```

Reply:
Thanks! https://github.com/primev/mev-commit/commit/2616c35b74a3ef82079b745128548c61744b1f84